frontend/public/components/edit-yaml.jsx

```diff
@@ -112,6 +113,20 @@ export const EditYAML_ = connect(stateToProps)(
         }
 
         createResources(objs, isDryRun) {
+          // If a namespace is being created, wait for each request synchronously.
+          if (hasNamespaceObj(objs)) {
+            async () => {
+              const promiseRequests = [];
+
+              objs.reduce(async (previousObj, currentObj) => {
+                await previousObj;
+                await promiseRequests.push(k8sCreate(this.getModel(currentObj), currentObj));
+              }, Promise.resolve([]));
+
+              return await promiseRequests;
+            };
```

Comment:
I think you can do something like this if you make `createResources` async

```ts
              const promises = [];
              for (const obj of objs) {
                const request = await k8sCreate(this.getModel(obj), obj);
                promises.push(request);
              }
              
              return promises;
```

Reply:
But you'll need error handling as well.

Reply:
Is this also code ever executed as written? It looks like an anonymous function that is never invoked.

Reply:
@TheRealJon An `async` function always returns a promise. For example:
```
async function fn() {
  return 'hello';
}
fn().then(console.log)
// hello
```
So I think the usual parenthesis` (---)() `for executing anonymous function is unnecessary in this case.

Reply:
> @TheRealJon An `async` function always returns a promise. For example:
> 
> ```
> async function fn() {
>   return 'hello';
> }
> fn().then(console.log)
> // hello
> ```
> 
> So I think the usual parenthesis`(---)()`for executing anonymous function is unnecessary in this case.
Oops! Never mind the example. It is a function definition.

Reply:
> @TheRealJon An `async` function always returns a promise. For example:
> 
> ```
> async function fn() {
>   return 'hello';
> }
> fn().then(console.log)
> // hello
> ```
> 
> So I think the usual parenthesis`(---)()`for executing anonymous function is unnecessary in this case.

@cyril-ui-developer I agree with the first part of this. Async functions will return Promises, but they still need to be invoked in order to do so.

Reply:
> But you'll need error handling as well.

@spadgett The error is handled here. 
https://github.com/openshift/console/blob/master/frontend/public/components/import-yaml-results.tsx#L116